torch_xla/csrc/ops/ops.cpp

```diff
@@ -415,6 +416,29 @@ NodePtr BroadcastTensors(tensorflow::gtl::ArraySlice<const Value> tensors) {
                    std::move(lower_fn), /*num_outputs=*/tensors.size());
 }
 
+NodePtr Norm(const Value& input, c10::optional<at::Scalar> p,
+             c10::optional<at::ScalarType> dtype, at::IntArrayRef dim,
+             bool keepdim) {
+  std::vector<xla::int64> dimensions(dim.begin(), dim.end());
+  if (dimensions.empty()) {
+    if (input.shape().rank() <= 2) {
+      dimensions = xla::util::Iota<xla::int64>(input.shape().rank());
+    } else {
+      dimensions.push_back(input.shape().rank() - 1);
+    }
+  }
+  if (!p.has_value() || p->toDouble() == 2.0) {
+    NodePtr square = input * input;
+    NodePtr result = MakeNode<Sum>(square, dimensions, keepdim, dtype);
+    return Sqrt(result);
+  }
+  if (p->toDouble() == 1.0) {
+    // Nuclear Norm. Needs trace.
+  }
+  // Generic sum(x^p)^(1/p) norms.
+  XLA_ERROR() << "Norm kind not implemented: " << *p;
```

Comment:
Shouldn't we re-route to CPU in ATen instead of failing here?

Reply:
PR is incomplete. Just added the general form. Waiting for the nuclear one for the trace.